corehq/apps/ota/views.py

```diff
@@ -32,6 +42,70 @@ def restore(request, domain, app_id=None):
     return get_restore_response(domain, couch_user, app_id, **get_restore_params(request))
 
 
+@json_error
+@login_or_digest_or_basic_or_apikey()
+def search(request, domain):
+    """
+    Accepts search criteria as GET params, e.g. "https://www.commcarehq.org/a/domain/phone/search/?a=b&c=d"
+    Returns results as a fixture with the same structure as a casedb instance.
+    """
+    criteria = dict(request.GET)
+    try:
+        case_type = criteria.pop('case_type')
+    except KeyError:
+        return HttpResponse('Search request must specify case type', status=400)
+
+    search_es = CaseSearchES()
+    search_es = search_es.domain(domain)
+    search_es = search_es.case_type(case_type)
+    config = CaseSearchConfig(domain=domain).config
+    fuzzies = config.get_fuzzy_properties_for_case_type(case_type)
+    for key, value in criteria.items():
+        search_es = search_es.case_property_query(key, value, fuzzy=(key in fuzzies))
+    results = search_es.values()
+    if should_use_sql_backend(domain):
+        cases = [get_instance_from_data(CommCareCaseSQLSerializer, result) for result in results]
+    else:
+        cases = [CommCareCase.wrap(flatten_result(result)) for result in results]
+    fixtures = CaseDBFixture(cases).fixture
+    return HttpResponse(fixtures, content_type="text/xml")
+
+
+@json_error
+@login_or_digest_or_basic_or_apikey()
+def claim(request, domain):
+    couch_user = CouchUser.from_django_user(request.user)
+    # TODO: use request.session to ensure only one extension case if multiple claims for the same beneficiary
+    #       come from the same AWW.
+    if request.method == 'POST':
+        host_id = request.POST['case_id']
+        host_type = request.POST.get('case_type')  # Nice to have,
+        host_name = request.POST.get('case_name')  # but optional
+        if not (host_type and host_name):
+            if should_use_sql_backend(domain):
+                case = CommCareCaseSQL.objects.get(case_id=host_id)
+            else:
+                case = CommCareCase.get(host_id)
+            host_type = case.type
+            host_name = case.name
+        claim_case_block = CaseBlock(
+            create=True,
+            case_id=uuid4().hex,
+            case_name=host_name,
+            case_type=CLAIM_CASE_TYPE,
+            owner_id=couch_user.user_id,
+            index={
+                'host': IndexAttrs(
+                    case_type=host_type,
+                    case_id=host_id,
+                    relationship='extension',
+                )
+            }
+        ).as_xml()
+        post_case_blocks([claim_case_block], {'domain': domain})
+    return HttpResponse(status=200)
```

Comment:
I think this will respond 200 when other request types hit this endpoint too. Also, are there other errors states that could occur here that the mobile would want to know about?